src/main/webapp/resources/js/apis/projects/project.js

```diff
@@ -0,0 +1,90 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+import { setBaseUrl } from "../../utilities/url-utilities";
+
+/**
+ * Redux API for all things project
+ * @type {Api<(args: (string | FetchArgs), api: BaseQueryApi, extraOptions: {}) => MaybePromise<QueryReturnValue<unknown, FetchBaseQueryError, FetchBaseQueryMeta>>, {updateProjectCoverage: *, getProjectDetails: *, getProjectCoverage: *, updateProjectDetails: *, updateProjectPriority: *, updateDefaultMetadataTemplate: *}, string, string, typeof coreModuleName> | Api<(args: (string | FetchArgs), api: BaseQueryApi, extraOptions: {}) => MaybePromise<QueryReturnValue<unknown, FetchBaseQueryError, FetchBaseQueryMeta>>, {updateProjectCoverage: *, getProjectDetails: *, getProjectCoverage: *, updateProjectDetails: *, updateProjectPriority: *, updateDefaultMetadataTemplate: *}, string, string, typeof coreModuleName | typeof reactHooksModuleName>}
+ */
+export const projectApi = createApi({
+  reducerPath: `projectsApi`,
+  baseQuery: fetchBaseQuery({
+    baseUrl: setBaseUrl(`ajax/project/details`),
+  }),
+  tagTypes: ["Project", "MetadataTemplate"],
+  endpoints: (build) => ({
+    /*
+    Get the default information about a project
+     */
+    getProjectDetails: build.query({
+      query: (projectId) => ({
+        url: "",
+        params: { projectId },
+      }),
+      providesTags: ["Project"],
+    }),
+    /*
+    Update details within a project
+     */
+    updateProjectDetails: build.mutation({
+      query: ({ projectId, field, value }) => ({
+        url: "",
+        params: { projectId },
+        body: { field, value },
+        method: "PUT",
+      }),
+      invalidatesTags: ["Project"],
+    }),
+    /*
+    Update the processing coverage
+     */
+    updateProjectCoverage: build.mutation({
+      query: ({ projectId, coverage }) => {
+        return {
+          url: "/coverage",
+          method: "PUT",
+          params: {
+            projectId,
+          },
+          body: coverage,
+        };
+      },
+      invalidatesTags: ["Project"],
+    }),
+    /*
+    Update the priority of the project pipelines
+     */
+    updateProjectPriority: build.mutation({
+      query: ({ projectId, priority }) => ({
+        url: "/priority",
+        method: "PUT",
+        params: {
+          projectId,
+          priority,
+        },
+      }),
+      invalidatesTags: ["Project"],
+    }),
+    /*
+    Update the current default metadata template for a project
+     */
+    updateDefaultMetadataTemplate: build.mutation({
+      query: ({ projectId, templateId }) => ({
+        url: "/set-project-default",
```

Comment:
Since this is now in the `ProjectDetailsAjaxController` we should update the url to `/set-project-default-template`

Reply:
Updated in 398c4bb